core/templates/dev/head/pages/exploration_editor/editor_tab/StateHints.js

```diff
@@ -51,6 +51,13 @@ oppia.controller('StateHints', [
     };
 
     $scope.changeActiveHintIndex = function(newIndex) {
+      var currentActiveIndex = $scope.activeHintIndex;
+      if (currentActiveIndex !== null && (
+          !stateHintsService.displayed[currentActiveIndex].hintText)) {
+        alertsService.addInfoMessage('Deleting empty hint.');
+        stateHintsService.displayed.splice(currentActiveIndex, 1);
+        stateHintsService.saveDisplayedValue();
+      }
     // If the current hint is being clicked on again, close it.
```

Comment:
OK, sounds good. Let me drop the lgtm for the time being while you do that, then :)